framework-support/jobrunr-spring-boot-2-starter/src/main/java/org/jobrunr/scheduling/AsyncJobPostProcessor.java

```diff
@@ -0,0 +1,49 @@
+package org.jobrunr.scheduling;
+
+import org.jobrunr.jobs.annotations.Job;
+import org.jobrunr.jobs.annotations.AsyncJob;
+import org.springframework.aop.framework.ProxyFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.InvocationTargetException;
+
+@Component
+public class AsyncJobPostProcessor implements BeanPostProcessor, BeanFactoryAware {
+    BeanFactory beanFactory;
+
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) {
+        if (bean.getClass().isAnnotationPresent(AsyncJob.class)) {
```

Comment:
I would suggest using `AnnotatedElementUtils.hasAnnotation(bean.getClass(), AsyncJob.class)`. That would take into account proxies, annotations on interfaces or even composed annotations if this is ever going to be used as such.

Reply:
Fixed